.github/workflows/release.yml

```diff
@@ -0,0 +1,72 @@
+name: devel-release
+on:
+  workflow_run:
+    workflows: [build-images]
+    types: [completed]
+    branches:
+      - main 
+
+permissions:
+  contents: write
+
+jobs:
+  release:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+       # Download all artifacts from the triggering run
+      - name: Download all artifacts from build run
+        uses: dawidd6/action-download-artifact@v11
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          run_id: ${{ github.event.workflow_run.id }}
+          path: dist
+
+      - name: Recreate devel release (replace assets)
+        run: |
+          set -euo pipefail
+          TAG=devel
+          gh release delete "${TAG}" --cleanup-tag --yes || true
+          gh release create "${TAG}" \
+            --title "Development Images" \
+            --notes "Automated build of host and guest images from main branch." \
+            --prerelease
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+      - name: Upload guest assets (default content type)
+        run: |
+          set -euo pipefail
+          # find guest files and upload via gh (preserves default Content-Type)
+          shopt -s globstar nullglob
+          guest_files=(dist/**/guest-*.efi)
+          if [ ${#guest_files[@]} -gt 0 ]; then
+            gh release upload devel "${guest_files[@]}" --clobber
+          else
+            echo "No guest assets found"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+      - name: Upload host assets (custom Content-Type)
+        run: |
+          set -euo pipefail
+          # Get release id to upload via raw uploads API so we can set a custom Content-Type.
+          RELEASE_ID=$(gh api repos/$GITHUB_REPOSITORY/releases/tags/devel --jq .id)
+          shopt -s globstar nullglob
+          for efi in dist/**/host-*.efi; do
+            fname=$(basename "$efi")
+            echo "Uploading host asset $fname with custom Content-Type"
+            curl -sS -X POST \
```

Comment:
You should definitely use `--fail` on the `curl` invocation. Otherwise you'll get silent failures.